src/js/media-container.js

```diff
@@ -19,58 +19,100 @@ template.innerHTML = `
     :host {
       box-sizing: border-box;
       position: relative;
-
-      /* Position controls at the bottom  */
-      display: inline-flex;
-      flex-direction: column-reverse;
+      display: inline-block;
 
       /* Max out at 100% width for smaller screens (< 720px) */
-      max-width: 100%;
+      // max-width: 100%;
       background-color: #000;
     }
 
+    *[part~=layer] {
+      position: absolute;
+      top: 0;
+      left: 0;
+      bottom: 0;
+      right: 0;
+      display: flex;
+      flex-flow: column nowrap;
+      align-items: stretch;
+      pointer-events: none;
+      background: none;
+    }
+
+    *[part~=gestures-layer] {
+      pointer-events: auto;
+    }
+
+    .spacer {
+      flex-grow: 1;
+      pointer-events: none;
+      background: none;
+    }
+
+    /* Position the media element to fill the container */
+    ::slotted([slot=media]) {
+      margin: auto 0 auto 0;
+      width: 100%;
+    }
+
     /* Video specific styles */
     :host(:not([audio])) {
-      height: 480px;
+      aspect-ratio: var(--media-aspect-ratio, auto 3 / 2);
       width: 720px;
     }
 
+    @supports not (aspect-ratio: 1 / 1) {
+      :host(:not([audio])) {
+        height: 480px;
+      }
+    }
+
     /* Safari needs this to actually make the element fill the window */
     :host(:-webkit-full-screen) {
       /* Needs to use !important otherwise easy to break */
       width: 100% !important;
       height: 100% !important;
     }
 
-    /* Position the media element to fill the container */
-    ::slotted([slot=media]) {
-      position: absolute;
-      top: 0;
-      left: 0;
-      width: 100%;
-      height: 100%;
-    }
-
     /* Hide controls when inactive and not paused and not audio */
     slot:not([media]) ::slotted() {
       opacity: 1;
       transition: opacity 0.25s;
       visibility: visible;
+      pointer-events: auto;
     }
 
     :host([user-inactive]:not([${MediaUIAttributes.MEDIA_PAUSED}]):not([audio])) slot:not([media]) ::slotted(*) {
       opacity: 0;
       transition: opacity 1s;
     }
-
-    slot:not([media]) ::slotted(media-control-bar)  {
-      width: 100%;
-    }
   </style>
-  <slot name="media"></slot>
-  <slot></slot>
+
+  <span part="layer media-layer">
+    <slot name="media"></slot>
+  </span>
+  <span part="layer gesture-layer">
+    <slot name="gestures-overlay"></slot>
+  </span>
+  <!--
+  <span part="layer text-tracks-layer">
```

Comment:
Could uncomment this if we want to anticipate our "let media-chrome optionally manage sub/cc rendering" option (though we'd still need to pin down some of the details before we could fully implement).